### ISSUE TITLE
Keras RNN Layer does not work with dynamic_decode

### ISSUE THREAD

**Issue by jmanly56** [tensorflow/tensorflow#26617]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:N/A
- TensorFlow installed from (source or binary):From pip
- TensorFlow version (use command below): tf-nightly-gpu-1.14.1.dev20190310
- Python version: 3.7.2
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10, 7.5 for CUDA Version 10.0
- GPU model and memory: GTX 1060 6GB

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
Using tf.keras.layers.RNN does not work in place of tf.nn.dynamic_rnn, or at least not in a way that I can find. Everytime I attempted to use it, I would get 
`Shape must be rank 2 but is rank 1 for 'decode/Training_Decoder/decoder/while/BasicDecoderStep/decoder/attention_wrapper/lstm_cell_3/MatMul_4' (op: 'MatMul') with input shapes: [1024], [1024,1024]`
but as soon as I swap to dynamic_rnn, it executes perfectly. However I see that dynamic_rnn will be removed. 
**Describe the expected behavior**
tf.keras.RNN should be equivalent to dynamic_rnn
**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
There was too much code so I used pastebin.
https://pastebin.com/iJ5diuie
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by ymodak** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by jmanly56** [comment]
My project was just a slightly modified version of this https://github.com/Currie32/Spell-Checker. 
`def encoding_layer(rnn_size, sequence_length, num_layers, rnn_inputs, keep_prob, direction):
    '''Create the encoding layer'''
    
    if direction == 1:
        with tf.name_scope("RNN_Encoder_Cell_1D"):
            for layer in range(num_layers):
                with tf.variable_scope('encoder_{}'.format(layer)):
                    lstm = tf.contrib.rnn.LSTMCell(rnn_size)

                    drop = tf.contrib.rnn.DropoutWrapper(lstm, 
                                                         input_keep_prob = keep_prob)

                    enc_output, enc_state = tf.nn.dynamic_rnn(drop, 
                                                              rnn_inputs,
                                                              sequence_length,
                                                              dtype=tf.float32)

return enc_output, enc_state`
I changed the dropout wrapper to both not being there, and also a keras dropout layer. I changed that cell to a keras LSTM cell. And tf.nn.dynamic_rnn was changed to tf.keras.RNN.

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
